File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.js
/**
 * @param {number[]} candidates
 * @return {number}
 */
var largestCombination = function(candidates) {
    const freq = Array(32).fill(0);
    var ans = 0
    const dp = (num) => {
        for (let i = 0; i < 32; ++i) {
            let m = 1 << i
            if (m & num)
                freq[i]++
            ans = Math.max(ans, freq[i])
        }
    }

    for(let num of candidates)
        dp(num)
    
    return ans
};

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/2356-largest-combination-with-bitwise-and-greater-than-zero.go
func largestCombination(candidates []int) int {
	freq := make([]int, 32)
	ans := 0
	for _, num := range candidates {
		for i := 0; i < 32; i++ {
		    mask := 1 << i
		    if mask & num != 0 {
			    freq[i]++
		    }
		    ans = max(ans, freq[i])
	    }
	}

	return ans
}